surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -44,32 +39,14 @@ public TestsToRun apply(List<String> orderStrategy) {
         return testsToRun;
     }
 
-    // TODO inverse creation of it - shouldn't belong to mvn package
     private TestExecutionPlanner getPlannerForStrategy(String orderStrategy) {
         final File projectDir = new File(System.getProperty("user.dir"));
-        final String previousCommit = System.getProperty("git.previous.commit", "HEAD");
-        final String commit = System.getProperty("git.commit", "HEAD");
-
         final String[] globPatternsAsArray = getGlobPatterns();
 
-        if (Objects.equals(orderStrategy, "new")) {
-            return new NewFilesDetector(projectDir, previousCommit, commit, globPatternsAsArray);
-        } else if (Objects.equals(orderStrategy, "changed")) {
-            return new ChangedFilesDetector(projectDir, previousCommit, commit, globPatternsAsArray);
-        } else if (Objects.equals(orderStrategy, "affected")) {
-            // TODO logic of inspecting git changes should be some where common so it is not recalculated several times
-            // TODO in fact there are at least two things to be put in a Context to be reused (git changes (main and test) and graph of dependencies between tests and main classes
-
-            // For now we recalculate everything
-            final Set<File> mainClasses =
-                new NewFilesDetector(projectDir, previousCommit, commit, "**/src/main/java/**/*.java").getFiles();
-
-            mainClasses.addAll(
-                new ChangedFilesDetector(projectDir, previousCommit, commit, "**/src/main/java/**/*.java").getFiles());
-
-            return new AffectedChangesDetector(new FileSystemTestClassDetector(projectDir), mainClasses);
-        } else if(Objects.equals(orderStrategy, "failed")) {
-            return new FailedTestsDetector();
+        for (final TestExecutionPlannerFactory testExecutionPlanner : ServiceLoader.load(TestExecutionPlannerFactory.class)) {
```

Comment:
the code `ServiceLoader.load(TestExecutionPlannerFactory.class)` will be called for every strategy:
https://github.com/arquillian/smart-testing/pull/7/files#diff-356cfd962197d56c7e69c381ba349281R26
but the list of the implementations won't change during the loop. I'm not sure how big is the load (if there is any), but I believe that it would be good to get the list of implementations before the loop and use it as a local variable.